test/sharness/t0180-pubsub.sh

```diff
@@ -10,83 +10,143 @@ test_expect_success 'init iptb' '
   iptb init -n $NUM_NODES --bootstrap=none --port=0
 '
 
-startup_cluster $NUM_NODES --enable-pubsub-experiment
+run_pubsub_tests() {
+  test_expect_success 'peer ids' '
+    PEERID_0=$(iptb get id 0) &&
+    PEERID_2=$(iptb get id 2)
+  '
+  
+  # ipfs pubsub sub
+  test_expect_success 'pubsub' '
+    echo "testOK" > expected &&
+    touch empty &&
+    mkfifo wait ||
+    test_fsh echo init fail
+  
+    # ipfs pubsub sub is long-running so we need to start it in the background and
+    # wait put its output somewhere where we can access it
+    (
+      ipfsi 0 pubsub sub --enc=ndpayload testTopic | if read line; then
```

Comment:
What is `ipfsi`? (I mean, where can I read some information about it?)

Reply:
It's a function defined in `test/sharness/lib/iptb-lib.sh`.